File: Project2_Java/src/pl/pg/oop/project2/organisms/plants/Sow_Thistle.java
package pl.pg.oop.project2.organisms.plants;

import pl.pg.oop.project2.organisms.OrganismsEnum;
import pl.pg.oop.project2.organisms.Plant;
import pl.pg.oop.project2.organisms.World;

import java.awt.*;

public class Sow_Thistle extends Plant {
    private final static int TRY_TO_SPREAD = 3;
    public Sow_Thistle(World world, Point position) {
        super(world, position);
        this.TypeOfOrganism = OrganismsEnum.SOW_THISTLE;
    }

    @Override
    public boolean action() {
        boolean Flag = false;
        for(int i = 0; i < TRY_TO_SPREAD; i++) {
            if(Flag) break;
            Flag = super.action();
        }
        return Flag;
    }

    @Override
    public String toString() {
        return "Sow_Thistle";
    }
}

File: Project3_Python/Plants/Sow_Thistle.py
import World
from Plant import Plant
from Organism import OrganismsEnum


class Sow_Thistle(Plant):
    __TRY_TO_SPREAD = 3

    def __init__(self, world, position):
        super().__init__(world, position)
        self.type_of_organism = OrganismsEnum.Sow_thistle

    def action(self):
        flag = False
        for i in range(self.__TRY_TO_SPREAD):
            if flag:
                break
            flag = super().action()
        return flag

    def __str__(self):
        return "Sow_Thistle"
